sdk/keyvault/azure-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java

```diff
@@ -0,0 +1,1453 @@
+     * stored certificate and attributes that are not specified in the request are left unchanged.. This operation requires the certificates/update permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Gets latest version of the certificate, changes its enabled status and then updates it in the Azure Key Vault. Prints out the
+     * returned certificate details when a response has been received.</p>
+     *
+     * {@codesnippet com.azure.security.keyvault.certificates.async.certificateclient.updateCertificateWithResponse#CertificateBase}
+     *
+     * @param certificate The {@link CertificateBase base certificate} object with updated properties.
+     * @throws NullPointerException if {@code certificate} is {@code null}.
+     * @throws ResourceNotFoundException when a certificate with {@link CertificateBase#name() name} and {@link CertificateBase#version() version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@link CertificateBase#name() name} or {@link CertificateBase#version() version} is empty string.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link CertificateBase updated certificate}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Certificate>> updateCertificateWithResponse(CertificateBase certificate) {
+       return withContext(context -> updateCertificateWithResponse(certificate, context));
+    }
+
+    Mono<Response<Certificate>> updateCertificateWithResponse(CertificateBase certificateBase, Context context) {
+        Objects.requireNonNull(certificateBase, "The certificate input parameter cannot be null");
+        CertificateUpdateParameters parameters = new CertificateUpdateParameters()
+            .tags(certificateBase.tags())
+            .certificateAttributes(new CertificateRequestAttributes(certificateBase));
+        return service.updateCertificate(endpoint, certificateBase.name(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE, context)
+            .doOnRequest(ignored -> logger.info("Updating certificate - {}",  certificateBase.name()))
+            .doOnSuccess(response -> logger.info("Updated the certificate - {}", certificateBase.name()))
+            .doOnError(error -> logger.warning("Failed to update the certificate - {}", certificateBase.name(), error));
+    }
+
+    /**
+     * Gets information about the certificate which represents {@link CertificateBase certificateBase} from the key vault. This
+     * operation requires the certificates/get permission.
+     *
+     * <p>The list operations {@link CertificateAsyncClient#listCertificates()} and {@link CertificateAsyncClient#listCertificateVersions(String)} return
+     * the {@link Flux} containing {@link CertificateBase base certificate} as output excluding the properties like secretId and keyId of the certificate.
```

Comment:
Do a general tidy up to not convert the type name `CertificateBase` into something else like `base certificate`, unless it makes sense in the context of the documentation. In these cases, it doesn't and should be removed.

Reply:
done